pkg/controller/change/network/watcher_test.go

```diff
@@ -154,51 +156,74 @@ func TestDeviceWatcher(t *testing.T) {
 		},
 	}
 
-	deviceClient := NewMockDeviceServiceClient(ctrl)
-	deviceClient.EXPECT().List(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, request *devicetopo.ListRequest) (devicetopo.DeviceService_ListClient, error) {
-		listClient := NewMockDeviceService_ListClient(ctrl)
-		listClient.EXPECT().Recv().Return(&devicetopo.ListResponse{
-			Type: devicetopo.ListResponse_NONE,
-			Device: &devicetopo.Device{
-				ID:      devicetopo.ID("device-1"),
-				Type:    "DeviceSim",
-				Version: "1.0.0",
-				Protocols: []*devicetopo.ProtocolState{
-					{
-						Protocol:          devicetopo.Protocol_GNMI,
-						ChannelState:      devicetopo.ChannelState_CONNECTED,
-						ConnectivityState: devicetopo.ConnectivityState_REACHABLE,
+	deviceClient := NewMockTopoClient(ctrl)
+	deviceClient.EXPECT().Subscribe(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, request *topo.SubscribeRequest) (topo.Topo_SubscribeClient, error) {
+		listClient := NewMockTopo_SubscribeClient(ctrl)
+		listClient.EXPECT().Recv().Return(&topo.SubscribeResponse{
+			Update: &topo.Update{
+				Type: topo.Update_UNSPECIFIED,
+				Object: &topo.Object{
+					ID:   topo.ID("device-1"),
+					Type: topo.Object_ENTITY,
+					Attributes: map[string]string{
+						topo.Type:    "DeviceSim",
```

Comment:
I think the type should come from the "kind"

Reply:
done